File: 2026/02-February/03-02-26/0509-fibonacci-number.cpp
/*
 * LeetCode 509 · Fibonacci Number · Easy
 * https://leetcode.com/problems/fibonacci-number/
 *
 * Pattern : Dynamic Programming
 * Solved  : 03 Feb 2026
 * Time    : O(n)
 * Space   : O(1)
 */

class Solution {
public:
    int fib(int n) {
        if(n==0) return n;
        if(n==1) return n;
        return (fib(n-1)+fib(n-2));
        
    }
};

File: 2026/09-September/23-09-26/0509-fibonacci-number.py
"""
LeetCode 509 · Fibonacci Number · Easy
https://leetcode.com/problems/fibonacci-number/

Pattern : Dynamic Programming
Solved  : 23 Sep 2026
Time    : O(n) · Runtime: 683 ms (Beats 7.4%)
Space   : O(1) · Memory: 12.5 MB (Beats 18.3%)
"""

class Solution(object):
    def fib(self, n):
        if n==0:
            return 0 
        if n==1:
            return 1
        return self.fib(n-1)+self.fib(n-2)
